mne/decoding/regressor.py

```diff
@@ -0,0 +1,139 @@
+# Authors: Romain Trachel <trachelr@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+
+class LinearRegressor():
+    """
+    This object clones a Linear Model from sklearn
+    and updates the attribute for each fit. The model coefficient
+    can be interpreted using the attribute patterns [1].
+
+    Parameters
+    ----------
+    reg : object | None
+        A linear regressor from sklearn with a fit method 
+        that updates a coef_ attribute.
+        If None the classifier will be a LinearRegressor
+
+    Attributes
+    ----------
+    filters_ : ndarray
+        If fit, the filters used to decompose the data, else None.
+    patterns_ : ndarray
+        If fit, the patterns used to restore M/EEG signals, else None.
+    
+    References
+    ----------
+    """
+    def __init__(self, reg):
+        if reg is None:
+            from sklearn.linear_model import LinearRegression
+            reg = LinearRegression()
+        
+        self.reg = reg
+    
+    def fit(self, X, y):
+        """Estimate the coeffiscient of the linear regressor.
+        Save the coeffiscient in the attribute filters_ and 
+        computes the attribute patterns_ using [1].
+
+        Parameters
+        ----------
+        X : array, shape=(n_epochs, n_features)
+            The data to estimate the coeffiscient.
+        y : array, shape=(n_epochs, n_target)
+            The target for each epoch.
+
+        Returns
+        -------
+        self : instance of LinearRegressor
+            Returns the modified instance.
+        
+        References
+        ----------
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("X should be of type ndarray (got %s)."
+                             % type(X))
+        if not isinstance(y, np.ndarray):
+            raise ValueError("y should be of type ndarray (got %s)."
+                             % type(y))
```

Comment:
y could be a list. In scikit-learn we support array-like and use the check function in validation.py